File: SmartPay/Models/SubCategory.cs
namespace SmartPay.Models;

public class SubCategory
{
    public int Id { get; set; }
    
    public string Name { get; set; }
    public string ImageUrl { get; set; }
    public int CategoryId { get; set; }
    public Category Category { get; set; }
    public List<Product> Products { get; set; }
}

public class SubCategoryViewModel
{
    public int Id { get; set; }
    
    public string Name { get; set; }
}


public class SubCategoryProductViewModel
{
    public int Id { get; set; }
    
    public string Name { get; set; }
    public string ImageUrl { get; set; }
    
    public CategoryViewModel Category { get; set; }
}

File: SmartPay/ClientApp/src/api/models/SubCategory.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { Category } from './Category';
import type { Product } from './Product';

export type SubCategory = {
    id?: number;
    name?: string | null;
    categoryId?: number;
    category?: Category;
    products?: Array<Product> | null;
};